core/src/main/java/org/elasticsearch/search/internal/InternalSearchHit.java

```diff
@@ -557,7 +564,9 @@ public void readFrom(StreamInput in) throws IOException {
         score = in.readFloat();
         id = in.readOptionalText();
         type = in.readOptionalText();
-        nestedIdentity = in.readOptionalStreamable(InternalNestedIdentity::new);
+        if (in.readBoolean()) {
```

Comment:
I want to introduce a ctor that takes a StreamInput, but to do that I need to read the bool that signals if the object was written outside. I changed this a bit furhter now so InternalNestedIdentity implements `Writeable`. I agree agree we generally don't want big changes here, but I think in terms of wire protocol this is equivalent and to me makes the code easier to read.